src/scripts/fetch-routes.ts

```diff
@@ -0,0 +1,187 @@
+import { locales } from "../lib/i18n";
+
+type RouteConfig = {
+  queryOperation: string;
+  path: string;
+};
+
+// these route are static and will not need any further processing
+const staticRoutesConfig = [
+  "/",
+  "/blog/",
+  "/services/",
+  "/contact/",
+  "/cases/",
+  "/jobs/",
+  "/work-at/",
+  "/lustrum/",
+  "/events/",
+  "/faq/",
+  "/energy-first/",
+];
+
+// these routes are generated dynamically by fetching all slugs for each page type
+// the slugs are appended to the value of the 'baseRoute' property
+const dynamicRoutesConfig: RouteConfig[] = [
+  {
+    queryOperation: "allServices",
+    path: "/services/",
+  },
+  {
+    queryOperation: "allBlogPosts",
+    path: "/blog/",
+  },
+  {
+    queryOperation: "allCaseItems",
+    path: "/cases/",
+  },
+  {
+    queryOperation: "allPeople",
+    path: "/team/",
+  },
+  {
+    queryOperation: "allEventItems",
+    path: "/events/",
+  },
+  {
+    queryOperation: "allContactConfirmations",
+    path: "/contact/",
+  },
+  {
+    queryOperation: "allJobs",
+    path: "/jobs/",
+  },
+];
+
+// fetches data from DatoCMS using the GraphQL API
+const fetchDatoQuery = async ({
+  query,
+  variables = {},
+}: {
+  query: string;
+  variables?: Record<string, unknown>;
+}) => {
+  return fetch("https://graphql.datocms.com/", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+      Accept: "application/json",
+      Authorization: process.env.DATO_API_TOKEN,
+    } as HeadersInit,
+    body: JSON.stringify({
+      query,
+      variables,
+    }),
+  })
+    .then((res) => res.json())
+    .then(({ data, errors }) => {
+      if (errors) {
+        console.log(errors);
+      }
+
+      return data;
+    });
+};
+
+// fetches a paginated list of slugs for a given operation
+const fetchPaginatedSlugsForOperation = async ({
+  operation,
+  locale,
+  skip,
+}: {
+  operation: string;
+  locale: string;
+  skip: number;
+}) => {
+  return fetchDatoQuery({
+    query: `
+                query ${operation}($skip: IntType, $locale: SiteLocale) {
+                    ${operation}(first: 100, skip: $skip, locale: $locale) {
+                        slug
+                    }
+                }
+            `,
+    variables: {
+      locale,
+      skip,
+    },
+  }).then((data) => data[operation]);
+};
+
+// fetches the total number of items for a given operation
+const fetchMetaForOperation = async ({
+  operation,
+  locale,
+}: {
+  operation: string;
+  locale: string;
+}) => {
+  return fetchDatoQuery({
+    query: `
+            query Meta ($locale: SiteLocale) {
+                _${operation}Meta(locale: $locale) {
+                    count
+                }
+            }
```

Comment:
Just like Selwyn mentioned in the other comment, indentation is a bit big here as well